boileroom/backend/apptainer.py

```
@@ -0,0 +1,724 @@
+        An available port number.
+
+    Raises
+    ------
+    RuntimeError
+        If no available port is found within max_attempts.
+    """
+    for port in range(start_port, start_port + max_attempts):
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            if sock.connect_ex(("127.0.0.1", port)) != 0:
+                return port
+    raise RuntimeError(f"Could not find available port in range {start_port}-{start_port + max_attempts - 1}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)